src/class-wpml-gutenberg-integration.php

```diff
@@ -120,9 +113,7 @@ public function string_translated(
 				$content .= $this->render_block( $block );
 			}
 
-			$this->sitepress->switch_lang( $lang );
-			wp_update_post( array( 'ID' => $translated_post_id, 'post_content' => $content ) );
-			$this->sitepress->switch_lang( null );
+			wpml_update_escaped_post( [ 'ID' => $translated_post_id, 'post_content' => $content ], $lang );
```

Comment:
With the new function `wpml_update_escaped_post` we can pass the language as a parameter so I could remove the `SitePress` dependency (here only for this purpose).